census/estimator/trainer/model.py

```diff
@@ -0,0 +1,185 @@
+        'marital_status', hash_bucket_size=100),
+    layers.sparse_column_with_hash_bucket(
+        'relationship', hash_bucket_size=100),
+    layers.sparse_column_with_hash_bucket(
+        'workclass', hash_bucket_size=100),
+    layers.sparse_column_with_hash_bucket(
+        'occupation', hash_bucket_size=1000),
+    layers.sparse_column_with_hash_bucket(
+        'native_country', hash_bucket_size=1000),
+
+    # Continuous base columns.
+    layers.real_valued_column('age'),
+    layers.real_valued_column('education_num'),
+    layers.real_valued_column('capital_gain'),
+    layers.real_valued_column('capital_loss'),
+    layers.real_valued_column('hours_per_week'),
+]
+
+
+def build_estimator(model_dir, embedding_size=8, hidden_units=None):
+  (gender, race, education, marital_status, relationship,
+   workclass, occupation, native_country, age,
+   education_num, capital_gain, capital_loss, hours_per_week) = INPUT_COLUMNS
+  """Build an estimator."""
+  # Sparse base columns.
+  # Reused Transformations.
+  age_buckets = layers.bucketized_column(
+      age, boundaries=[18, 25, 30, 35, 40, 45, 50, 55, 60, 65])
+
+  # Wide columns and deep columns.
+  wide_columns = [
+      layers.crossed_column(
+          [education, occupation], hash_bucket_size=int(1e4)),
+      layers.crossed_column(
+          [age_buckets, race, occupation], hash_bucket_size=int(1e6)),
+      layers.crossed_column(
+          [native_country, occupation], hash_bucket_size=int(1e4)),
+      gender,
+      native_country,
+      education,
+      occupation,
+      workclass,
+      marital_status,
+      relationship,
+      age_buckets,
+  ]
+
+  deep_columns = [
+      layers.embedding_column(workclass, dimension=embedding_size),
+      layers.embedding_column(education, dimension=embedding_size),
+      layers.embedding_column(marital_status, dimension=embedding_size),
+      layers.embedding_column(gender, dimension=embedding_size),
+      layers.embedding_column(relationship, dimension=embedding_size),
+      layers.embedding_column(race, dimension=embedding_size),
+      layers.embedding_column(native_country, dimension=embedding_size),
+      layers.embedding_column(occupation, dimension=embedding_size),
+      age,
+      education_num,
+      capital_gain,
+      capital_loss,
+      hours_per_week,
+  ]
+
+  return tf.contrib.learn.DNNLinearCombinedClassifier(
+      model_dir=model_dir,
+      linear_feature_columns=wide_columns,
+      dnn_feature_columns=deep_columns,
+      dnn_hidden_units=hidden_units or [100, 70, 50, 25])
+
+
+def is_sparse(column):
+  return isinstance(column, layers.feature_column._SparseColumn)
+
+
+def feature_columns_to_placeholders(feature_columns, default_batch_size=None):
+    return {
+        column.name: tf.placeholder(
+            tf.string if is_sparse(column) else tf.float32,
+            [default_batch_size]
+        )
+        for column in feature_columns
+    }
+
+
+def json_serving_input_fn():
```

Comment:
See below, on why I disagree that serving time input should be the same as training time. They are totally different use cases.